plugins/modules/ec2_instance.py

```diff
@@ -276,6 +276,28 @@
     description:
       - The placement group that needs to be assigned to the instance
     type: str
+  metadata_options:
```

Comment:
After thinking about this some more, I believe we should go with the names/options that are used and returned by boto3 (which should match what a user would type on the cli). Which would make the suboptions here:
```
    metadata_options:
        http_tokens: 'optional'|'required',
        http_endpoint: 'disabled'|'enabled'
```

Otherwise this PR looks, good, thanks @mandar242!